### DISCUSSION TITLE
How to print a list of chosen add-ons on the PDF ticket?

### DISCUSSION THREAD

**Discussion by c0de-bender** [pretix/pretix#3702] [Q&A] [answered]
Hi,

_tl;dr:_ The description of `Generate tickets for add-on products` on https://docs.pretix.eu/en/latest/user/events/tickets.html states:

> ... If disabled, you can still print a list of chosen add-ons e.g. on the PDF tickets.

Does it mean I can configure something so that this list is printed there automatically? If yes, how can I make that happen?

Alternative solution for us: Is there a way to export a (spreadsheet) list of paid orders which contains one line per each main product and lists all the associated add-on products in that line (in one or multiple columns)? I can probably easily write an SQL query to get this, but I'd be happy not having to do it, if Pretix can do this already.

### Background

We use Pretix for an event with 3 add-on products:

  - sleeping spot
  - additional donation (donaters receive a small gift)
  - student discount

There'll be one central check-in desk and we want to make sure to check for all these add-ons – basically they all "require special attention" at the entrance (tell people where to sleep, hand over gift, check student card).

We'd like to be able to only scan or look at the main ticket PDF and thereby gather knowledge about all booked add-ons, because:

  - we don't want the attendees need to remember about their booked add-ons,
  - we don't want them to have to show 4 QR codes (inconvenient and slow),
  - we still want to give them their gift and necessary information,
  - we want to be able to check their student card. This might have been better as a variation of a main product, but there are some reasons why we like the add-on approach more.

Hence the above question. It would be enough for us to look at a PDF and see all the add-ons, because we just need to know the person arrived.

Note: Ideally, I'd like to only scan the main ticket's QR code and be notified by pretixSCAN about all associated add-ons of this main ticket. I've seen quite some questions about this (e.g. #1331, #2433, #2539, #3526) and understood that it's not possible yet (but [planned for the near future?!](https://github.com/pretix/pretix/discussions/3526#discussioncomment-6789332)). Would be great to have it in the future!

### A word of gratitude

Thanks a lot for putting so much effort in Pretix! It's a real pleasure to go through the settings and configure our event, discovering cool features and functions that make our work easier! I like how intuitive everything is! Whenever things are not clear to me, the documentation is usually pretty good and helpful, or you as Pretix team help out quite quickly by replying to questions here. That's awesome and it's great that you open-sourced this for anyone to freely use it!

I'll try to give something back in return by providing feedback, suggesting improvements, reporting some bugs and maybe even fixing some of them as soon as I have more time!

**Comment by c0de-bender** [comment, marked as answer]
I found it myself! I didn't have a close enough look at the PDF ticket layout editor – it's included there! And I must say: This editor is super awesome!

For anyone else also looking for this, here is how to get there (assuming the PDF ticket plugin is active): In the event backend go to `Settings > Tickets`, then under `Download Format > PDF output` click on `Open Layout Designer`. There I could add a text item (right side) and then in the content field choose "List of Add-Ons" (next to a long list of other placeholders, that's amazing!). This will print a list of the chosen add-on's names!
Note: I could even add a new object "Other QR code" and choose "List of Add-Ons" there, too. But: this will not be a QR code scannable by pretixSCAN, but just a QR code containing the exact same text(!) as the text object "List of Add-Ons".

However, I realized a major drawback of my plan to use this printed list for check-in: We let PDF tickets be generated after payment receival and then the add-on list will be in the (first) generated ticket PDF. When – afterwards – an add-on ticket gets cancelled (by the user or us), the user can still come to our check-in with the first PDF. The main ticket QR code will show as valid in pretixSCAN (as expected), but we cannot check if all the products on the add-ons list are valid just by looking at the list on their ticket. So this is not an option in the way I hoped for. We'll probably just go with the paper list solution.

I'm still amazed by the awesome ticket editor! And for next year, we hope that the "one QR code for all tickets" feature will be ready to use :)

Thanks so much for this amazing software!

**Reply by raphaelm** [reply]
Thanks for the kind words! Yes, this is an area of functionality with so many different use cases that it is hard to find one-size-fits-all solutions. There will absolutely be some more improvements in the SCAN area, though :)
